api-module/src/main/java/org/devridge/api/domain/sociallogin/context/OAuth2MemberInfoContext.java

```diff
@@ -0,0 +1,61 @@
+package org.devridge.api.domain.sociallogin.context;
+
+import org.devridge.api.domain.sociallogin.dto.response.oauth.OAuth2TokenResponse;
+import org.devridge.api.domain.sociallogin.entity.GoogleMember;
+import org.devridge.api.domain.sociallogin.entity.KakaoMember;
+import org.devridge.api.domain.sociallogin.entity.NaverMember;
+import org.devridge.api.domain.sociallogin.entity.OAuth2Member;
+import org.devridge.api.domain.sociallogin.strategy.member.GoogleMemberInfoProvider;
+import org.devridge.api.domain.sociallogin.strategy.member.KakaoMemberInfoProvider;
+import org.devridge.api.domain.sociallogin.strategy.member.NaverMemberInfoProvider;
+import org.devridge.api.domain.sociallogin.strategy.member.OAuth2MemberInfoStrategy;
+import org.devridge.api.exception.sociallogin.SocialLoginException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class OAuth2MemberInfoContext {
+
+    private final Map<String, OAuth2MemberInfoStrategy> strategies;
+    private final Map<String, Function<Map<String, Object>, OAuth2Member>> memberInfoProviderMap = new HashMap<>();
+
+    @Autowired
+    public OAuth2MemberInfoContext(
+            NaverMemberInfoProvider naverMemberInfoProvider,
+            GoogleMemberInfoProvider googleMemberInfoProvider,
+            KakaoMemberInfoProvider kakaoMemberInfoProvider
+    )
+    {
+        this.strategies = new HashMap<>();
+
+        strategies.put("naver", naverMemberInfoProvider);
```

Comment:
이렇게 바꿔보시는 건 어떤가요? 불변 객체를 만들어줄 수 있습니다.

```suggestion
        strategies = Map.of(
            "naver", naverMemberInfoProvider,
            "google", googleMemberInfoProvider,
            "kakao", kakaoMemberInfoProvider
    );
```

Reply:
오 감사합니다. 저렇게 바꿀게요